src/analysis/retail/warrior/fury/modules/spells/SlayerExecute.tsx

```diff
@@ -0,0 +1,193 @@
+    this.hasAshenJuggernaut = this.selectedCombatant.hasTalent(talents.ASHEN_JUGGERNAUT_TALENT);
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER).spell(SPELLS.EXECUTE_FURY), this.onCast);
+
+    this.addEventListener(
+      Events.applydebuff.by(SELECTED_PLAYER).spell(SPELLS.MARKED_FOR_EXECUTION),
+      this.onMarkedForExecutionStackChange,
+    );
+    this.addEventListener(
+      Events.applydebuffstack.by(SELECTED_PLAYER).spell(SPELLS.MARKED_FOR_EXECUTION),
+      this.onMarkedForExecutionStackChange,
+    );
+    this.addEventListener(
+      Events.removedebuff.by(SELECTED_PLAYER).spell(SPELLS.MARKED_FOR_EXECUTION),
+      this.onMarkedForExecutionStackChange,
+    );
+    this.addEventListener(
+      Events.removedebuffstack.by(SELECTED_PLAYER).spell(SPELLS.MARKED_FOR_EXECUTION),
+      this.onMarkedForExecutionStackChange,
+    );
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_FURY_TALENT_BUFF),
+      this.onSuddenDeathStackChange,
+    );
+    this.addEventListener(
+      Events.refreshbuff.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_FURY_TALENT_BUFF),
+      this.onSuddenDeathStackChange,
+    );
+    this.addEventListener(
+      Events.removebuffstack.by(SELECTED_PLAYER).spell(SPELLS.SUDDEN_DEATH_FURY_TALENT_BUFF),
+      this.updateRemoveBuffStack,
+    );
+  }
+
+  get suggestionThresholds() {
+    return {
+      actual: this.overusedExecutes,
+      isGreaterThan: {
+        minor: 0,
+        average: 5,
+        major: 12,
+      },
+      style: ThresholdStyle.NUMBER,
+    };
+  }
+
+  onMarkedForExecutionStackChange(
+    event: ApplyDebuffEvent | ApplyDebuffStackEvent | RemoveDebuffStackEvent | RemoveDebuffEvent,
+  ) {
+    this.markedForExecutionStacks = currentStacks(event);
+  }
+
+  updateRemoveBuffStack(event: RemoveBuffStackEvent) {
+    this.lastSuddenDeathRemoveBuffStackTimestamp = event.timestamp;
+  }
+
+  onSuddenDeathStackChange(event: ApplyBuffEvent | RefreshBuffEvent) {
+    // 2->1 stacks triggers refresh buff event, but doesn't extend duration
+    // so ignore those
+    // but we still want to listen to refreshbuff because
+    // 2->2 stacks triggers a refresh event and extends duration
+    if (
+      event.type === EventType.RefreshBuff &&
+      event.timestamp - this.lastSuddenDeathRemoveBuffStackTimestamp < 100
+    ) {
+      return;
+    }
+    this.suddenDeathExpirationTimestamp = event.timestamp + SUDDEN_DEATH_DURATION;
+  }
+
+  onCast(event: CastEvent) {
+    let usedForAshenJuggernaut = false;
+    let usedForSuddenDeath = false;
+    let usedForMarkedForExecution = false;
+
+    if (this.ashenJuggernautExpirationTimestamp - event.timestamp < BUFF_REFRESH_BUFFER) {
+      usedForAshenJuggernaut = true;
+    }
+    // Assume new AJ duration on cast of Execute
+    // which is true unless the execute is parried
+    this.ashenJuggernautExpirationTimestamp = event.timestamp + ASHEN_JUGGERNAUT_DURATION;
```

Comment:
making a note for future improvement that this is a case where an event link could be useful. by linking the execute to its damage, you can look ahead and check if it was parried so you don't need to assume.

Reply:
Agreed, will look into adding this in future updates (which will be coming for 11.1 since they're changing how RA works again)